src/guacscanner/guacscanner.py

```diff
@@ -0,0 +1,764 @@
+  --rdp-password=PASSWORD  If specified then the specified value will be used for the RDP password.  Otherwise, the password will be read from a local file.
+  --rdp-password-file=FILENAME  The file from which the RDP password will be read. [default: /run/secrets/rdp-password]
+  --rdp-username=USERNAME  If specified then the specified value will be used for the RDP username.  Otherwise, the username will be read from a local file.
+  --rdp-username-file=FILENAME  The file from which the RDP username will be read. [default: /run/secrets/rdp-username]
+  --region=REGION  The AWS region in which the VPC specified by --vpc-id exists.  Unused if --vpc-id is not specified. [default: us-east-1]
+  --sleep=SECONDS  Sleep for the specified number of seconds between executions of the Guacamole connection update loop. [default: 60]
+  --vnc-password=PASSWORD  If specified then the specified value will be used for the VNC password.  Otherwise, the password will be read from a local file.
+  --vnc-password-file=FILENAME  The file from which the VNC password will be read. [default: /run/secrets/vnc-password]
+  --vnc-username=USERNAME  If specified then the specified value will be used for the VNC username.  Otherwise, the username will be read from a local file.
+  --vnc-username-file=FILENAME  The file from which the VNC username will be read. [default: /run/secrets/vnc-username]
+  --vpc-id=VPC_ID        If specified then query for EC2 instances created
+                         or destroyed in the specified VPC ID.  If not
+                         specified then the ID of the VPC in which the host
+                         resides will be used.
+"""
+
+
+# Standard Python Libraries
+import datetime
+import logging
+import re
+import sys
+import time
+
+# Third-Party Libraries
+import boto3
+import docopt
+from ec2_metadata import ec2_metadata
+import psycopg
+from psycopg import sql
+from schema import And, Optional, Or, Schema, SchemaError, Use
+
+from ._version import __version__
+
+# TODO: Add exception handling for all the database accesses and
+# wherever else it is appropriate.  guacscanner currently just bombs
+# out if an exception is thrown, but it would probably make more sense
+# to print an error message and keep looping, keepin' the train
+# a-chooglin'.  See cisagov/guacscanner#5 for more details.
+
+# TODO: Create command line options with defaults for these variables.
+# See cisagov/guacscanner#2 for more details.
+DEFAULT_ADD_INSTANCE_STATES = [
+    "running",
+]
+DEFAULT_POSTGRES_DB_NAME = "guacamole_db"
+DEFAULT_POSTGRES_HOSTNAME = "postgres"
+DEFAULT_POSTGRES_PORT = 5432
+DEFAULT_REMOVE_INSTANCE_STATES = [
+    "terminated",
+]
+DEFAULT_AMI_SKIP_REGEXES = [
+    re.compile(r"^guacamole-.*$"),
+    re.compile(r"^nessus-.*$"),
+    re.compile(r"^samba-.*$"),
+]
+
+# Some precompiled regexes
+INSTANCE_ID_REGEX = re.compile(r"^.* \((?P<id>i-[0-9a-f]{17})\)$")
```

Comment:
This might benefit from a succinct comment just to mention the `(?P<value>)` usage.

Reply:
Please enjoy commit c5f57d1.